### PR TITLE
Feat: add data source for getting a list of Cloud Credentials

### LINKED ISSUE #358: Add data source for getting a list of Cloud Credentials
Add a new data source to list all Cloud Credentials in the org.
This should provide a list of all the credential names in a given Org filtered by a credential type.

This list should be able to be used to create a credential data source based on the names returned.

Usage example:

data "env0_cloud_credentials" "azure_deployment_credentials" {
credential_type = "AZURE_SERVICE_PRINCIPAL_FOR_DEPLOYMENT "
}

data "env0_azure_credentials" "azure_deployment_credentials" {
for_each = toset(data.env0_cloud_credentials.azure_deployment_credentials.names)
name = each.value
}

Relevant API:
    https://developer.env0.com/docs/api/b3A6ODc4NjAxMA-list-organization-credentials

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #358 

### Solution
Added the new data source.
Added acceptance tests.
Added integration tests.
Added example.

### CODE REVIEW

File: env0/data_cloud_credentials.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type:        schema.TypeString,
+					Description: "the credential name",
+				},
+			},
+			"credential_type": {
+				Type:             schema.TypeString,
+				Description:      "the type of cloud credential to filter by " + allowedCredentialTypesStr,
+				Optional:         true,
```
Comment (TomerHeber) [inline env0/data_cloud_credentials.go:42]: setting to optional incases where someone (for some reason) wants to get all the credentials regardless of type.

File: tests/integration/024_cloud_credentials/conf.tf [hunk 2]
```diff
@@ -0,0 +1,15 @@
+terraform {
+  backend "local" {
+  }
+  required_providers {
+    env0 = {
+      source = "terraform-registry.env0.com/env0/env0"
+    }
```
Comment (yaronya) [inline tests/integration/024_cloud_credentials/conf.tf:7]: Is this a new thing or am I wrong? I mean, shouldn't the integration tests use the local version and not the version from the public registry?

Comment (TomerHeber) [reply tests/integration/024_cloud_credentials/conf.tf:7]: yes. This is my bad.

Comment (TomerHeber) [reply tests/integration/024_cloud_credentials/conf.tf:7]: @yaronya - actually looking at it again. This is not new. All tests are like this.
I guess it's somehow working as expected...

Comment (avnerenv0) [reply tests/integration/024_cloud_credentials/conf.tf:7]: This is not the real registry (`terraform-registry.env0.com` doesn't exist).
This is set up in `buildFakeTerraformRegistry` in `harness.go`

**Review by avnerenv0 — APPROVED** [review]
👍

**Pull request merged**